include/solo5/mft_abi.h

```diff
@@ -99,55 +100,76 @@ struct mft {
 };
 
 /*
- * Manifest ELF note headers.
+ * Maximum supported number of manifest entires.
+ */
+#define MFT_MAX_ENTRIES 64
+
+/*
+ * HERE BE DRAGONS.
  *
- * Note that somewhat counter-intuitively, in practice the note section header
- * fields (namesz/descsz/type) are 32-bit words even on ELF64.
+ * The following structures and macros are used to declare a Solo5 "MFT1"
+ * format note at link time. This is somewhat tricky, as we need to ensure all
+ * structures are aligned correctly.
  */
-#define SOLO5_NOTE_NAME     "Solo5"
-#define SOLO5_NOTE_MANIFEST 0x3154464d /* "MFT1" */
+#define SOLO5_NOTE_NAME      "Solo5"
+#define SOLO5_NOTE_TYPE_MFT1 0x3154464d /* "MFT1" */
 
-struct mft_note_header {
-    uint32_t namesz;
-    uint32_t descsz;
-    uint32_t type;
-    char name[(sizeof(SOLO5_NOTE_NAME) + 3) & -4];
+/*
+ * Defines an Elf64_Nhdr with n_name filled in and padded to a 4-byte boundary,
+ * i.e. the common part of a Solo5-owned Nhdr.
+ */
+struct mft1_nhdr {
+    uint32_t n_namesz;
+    uint32_t n_descsz;
+    uint32_t n_type;
+    char n_name[(sizeof(SOLO5_NOTE_NAME) + 3) & -4];
 };
 
-struct mft_note {
-    struct mft_note_header h;
+/*
+ * Defines the entire note (header, descriptor content).
+ */
+struct mft1_note {
+    struct mft1_nhdr h;
+    /*
+     * (m) will be aligned to a MFT_NOTE_ALIGN boundary.
+     */
     struct mft m;
 };
```

Comment:
```suggestion
 };
_Static_assert(offsetof(struct mft1_note, m) == MFT1_NOTE_ALIGN,
        "(m) should be aligned to a MFT1_NOTE_ALIGN boundary");
```
Then we can get rid of the comment as well and ensure it statically, but it would require moving the definition of `MFT1_NOTE_ALIGN` above this struct definition.

Reply:
> Then we can get rid of the comment as well and ensure it statically, but it would require moving the definition of MFT1_NOTE_ALIGN above this struct definition.

I don't follow, what does this gain us?

Reply:
It would statically ensure what is currently asserted in the comment?

Reply:
To do that we want the following, placed after the definition of MFT1_NOTE_ALIGN:

```
_Static_assert((offsetof(struct mft1_note, m) & (MFT1_NOTE_ALIGN - 1)) == 0,
        "struct mft1_note.m is not aligned to a MFT1_NOTE_ALIGN boundary");
```

Right?

Reply:
Yes, that would do the trick!